### DISCUSSION TITLE
How to remove CORS header

### DISCUSSION THREAD

**Discussion by alexandereneroth** [vercel/next.js#37612] [Help] [answered]
Hi!
I have a static site ( nextjs project built with `next export`) running on Vercel.

For some reason all files served from the webserver are getting the header `Access-Control-Allow-Origin: *`. 
A security firm has flagged that as a potential vulnerability so i want to remove it.

How can i do that?

Thank you for reading!

**Comment by icyJoseph** [comment, marked as answer]
~~I am not sure if these reflect in `next export` but you can set headers for all routes on `next.config.js`. Alternatively~~ you can also do it through a `vercel.json` file. Please read more here:

https://vercel.com/support/articles/how-to-enable-cors#enabling-cors-using-vercel.json

Setting headers, redirects and rewrites is not supported when doing `next export`.

**Comment by alexandereneroth** [comment]
Thanks for your suggestion @icyJoseph 
Unfortunately i already tried setting Access-Control-Allow-Origin to null in vercel.json, but that made the site not build at all on Vercel (Vercel did not even pick up the commit from github, and github reported that the Vercel build failed)

**Reply by icyJoseph** [reply]
Why `null` though? It should be the domain of where you host the application. If I host on `https://example.com`, then that should be the value of the header. At least, I think that's where the security firm is coming from.

**Reply by alexandereneroth** [reply]
Thanks, I didn't think of that 🌟 . I wanted to remove the header, but I guess setting it to my own domain should work just as well.

**Reply by tolchai** [reply]
@alexandereneroth hi, i am struggling with the same issue, did you eventually found a solution for this? I tried to set the domain as value in Access-Control-Allow-Origin, but to no avail. Website is hosted on vercel.
